### PR TITLE
[libc] adjust linux's `mman.h` definitions

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
Use definitions from `<linux/mman.h>` to dispatch arch-specific flag values.
For example, `MCL_CURRENT/MCL_FUTURE/MCL_ONFAULT` are different on different architectures.

### CODE REVIEW

File: libc/include/llvm-libc-macros/sys-mman-macros.h [hunk 1]
```diff
@@ -9,8 +9,40 @@
...
+// Use definitions from <linux/mman.h> to dispatch arch-specific flag values.
+// For example, MCL_CURRENT/MCL_FUTURE/MCL_ONFAULT are different on different
+// architectures.
+#if __has_include(<linux/mman.h>)
+#include <linux/mman.h>
+#else
+#error "cannot use <sys/mman.h> without proper system headers."
+#endif
+
+// Some posix standard flags may not be defined in system headers.
```
Comment (michaelrj-google) [inline libc/include/llvm-libc-macros/sys-mman-macros.h:21]: are these still necessary?

File: libc/include/llvm-libc-macros/linux/CMakeLists.txt [hunk 2]
```diff
@@ -16,12 +16,6 @@ add_header(
     sys-ioctl-macros.h
 )
 
-add_header(
```
Comment (michaelrj-google) [inline libc/include/llvm-libc-macros/linux/CMakeLists.txt:19]: does this also need to be removed from the cmakelists for the other sys-mman-macros.h?

File: libc/include/llvm-libc-macros/sys-mman-macros.h (hunk 1 above)
Comment (SchrodingerZhu) [reply libc/include/llvm-libc-macros/sys-mman-macros.h:21]: Yes, `MAP_FAILED` and `POSIX_*` are not provided in UAPI.

File: libc/include/llvm-libc-macros/linux/CMakeLists.txt (hunk 2 above)
Comment (SchrodingerZhu) [reply libc/include/llvm-libc-macros/linux/CMakeLists.txt:19]: I suppose it is automatically detected inside `add_macro_header`:
```cmake
if(TARGET libc.include.llvm-libc-macros.${LIBC_TARGET_OS}.${name})
    add_header(
      ${name}
      HDR
        ${MACRO_HEADER_HDR}
      DEPENDS
        .${LIBC_TARGET_OS}.${name}
        ${MACRO_HEADER_DEPENDS}
    )
  else()
    add_header(
      ${name}
      HDR
        ${MACRO_HEADER_HDR}
      DEPENDS
        ${MACRO_HEADER_DEPENDS}
    )
  endif()
```

File: libc/include/llvm-libc-macros/sys-mman-macros.h (hunk 1 above)
Comment (SchrodingerZhu) [reply libc/include/llvm-libc-macros/sys-mman-macros.h:21]: https://elixir.bootlin.com/linux/latest/A/ident/MAP_FAILED

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Pull request merged**